libexec/tfenv-install

```diff
@@ -82,49 +82,64 @@ if [ -f "${dst_path}/terraform" ]; then
   exit 0;
 fi;
 
-# Add support of ARM64 for Linux & Apple Silicon
-case "$(uname -m)" in
-  aarch64* | arm64*)
-    # There is no arm64 support for versions:
-    # < 0.11.15
-    # >= 0.12.0, < 0.12.30
-    # >= 0.13.0, < 0.13.5
-    if [[ "${version}" =~ 0\.(([0-9]|10))\.\d* ||
-          "${version}" =~ 0\.11\.(([0-9]|1[0-4]))$ ||
-          "${version}" =~ 0\.12\.(([0-9]|[1-2][0-9]))$ ||
-          "${version}" =~ 0\.13\.[0-4]$
-    ]]; then
-      TFENV_ARCH="${TFENV_ARCH:-amd64}";
-    else
-      TFENV_ARCH="${TFENV_ARCH:-arm64}";
-    fi;
-    ;;
-  *)
-    TFENV_ARCH="${TFENV_ARCH:-amd64}";
-    ;;
-esac;
-
 case "$(uname -s)" in
   Darwin*)
-    os="darwin_${TFENV_ARCH}";
+    kernel="darwin";
     ;;
   MINGW64*)
-    os="windows_${TFENV_ARCH}";
+    kernel="windows";
     ;;
-  MSYS_NT*)
-    os="windows_${TFENV_ARCH}";
+  MSYSNT*)
+    kernel="windows";
     ;;
-  CYGWIN_NT*)
-    os="windows_${TFENV_ARCH}";
+  CYGWINNT*)
+    kernel="windows";
     ;;
   FreeBSD*)
-    os="freebsd_${TFENV_ARCH}";
+    kernel="freebsd";
     ;;
   *)
-    os="linux_${TFENV_ARCH}";
+    kernel="linux";
     ;;
 esac;
 
+# Add support of ARM64 for Linux & Apple Silicon
+case "$(uname -m)" in
+  aarch64* | arm64*)
+    case "${kernel}" in
+      "linux")
+        # There is no arm64 support for versions:
+        # < 0.11.15
+        # >= 0.12.0, < 0.12.30
+        # >= 0.13.0, < 0.13.5
+        if [[ "${version}" =~ 0\.(([0-9]|10))\.\d* ||
+              "${version}" =~ 0\.11\.(([0-9]|1[0-4]))$ ||
+              "${version}" =~ 0\.12\.(([0-9]|[1-2][0-9]))$ ||
```

Comment:
@Zordrak @denizgenc Unfortunately, this is still causing trouble:

```console
➜ tfenv use min-required
No installed versions of terraform matched '0.12.31:^0.12.31$'. Trying to install a matching version since TFENV_AUTO_INSTALL=true
Installing Terraform v0.12.31
Downloading release tarball from https://releases.hashicorp.com/terraform/0.12.31/terraform_0.12.31_darwin_arm64.zip
curl: (22) The requested URL returned error: 404

Tarball download failed
Installing a matching version failed
```

[`0.12.31`](https://releases.hashicorp.com/terraform/) is indeed the very last backport release caused by the keyring rotation.

```suggestion
              "${version}" =~ 0\.12\.(([0-9]|[1-3][0-9]))$ ||
```